pkgs/os-specific/linux/hostapd/default.nix

```diff
@@ -38,28 +56,54 @@ stdenv.mkDerivation rec {
     CONFIG_EAP_IKEV2=y
     CONFIG_EAP_TNC=y
     CONFIG_EAP_EKE=y
-    CONFIG_RADIUS_SERVER=y
-    CONFIG_IEEE80211R=y
-    CONFIG_IEEE80211N=y
-    CONFIG_IEEE80211AC=y
-    CONFIG_FULL_DYNAMIC_VLAN=y
-    CONFIG_VLAN_NETLINK=y
+
     CONFIG_TLS=openssl
     CONFIG_TLSV11=y
     CONFIG_TLSV12=y
-    CONFIG_INTERNETWORKING=y
+
+    CONFIG_SAE=y
+    CONFIG_SAE_PK=y
+
+    CONFIG_OWE=y
+    CONFIG_OCV=y
+
+    # TKIP is considered insecure and upstream support will be removed in the future
+    CONFIG_NO_TKIP=y
+
+    # Misc
+    CONFIG_PKCS12=y
```

Comment:
CONFIG_PKCS12=y
CONFIG_IPV6=y
CONFIG_RSN_PREAUTH=y
CONFIG_DPP=y
CONFIG_DPP2=y
are already enabled by default:
https://w1.fi/cgit/hostap/tree/hostapd/defconfig
Also CONFIG_OWE, CONFIG_WNM are experimental

Reply:
There seems to be a more fundamental question on how we want to handle this. Do we only want to track differences from upstream defconfig, or do we want to explicitly list all options that were considered?

>Also CONFIG_OWE, CONFIG_WNM are experimental
CONFIG_OWE is necessary for WPA3 compatibility, we shouldn't turn that off. Disabling WNM by default is a good idea because it is an incomplete implementation.

Reply:
> CONFIG_IPV6=y
> CONFIG_RSN_PREAUTH=y

These aren't in 2.10: https://w1.fi/cgit/hostap/tree/wpa_supplicant/defconfig?h=hostap_2_10

Reply:
I think rebasing on 2.10 is a fine choice.

Reply:
Done.